ros/angel_utils/python/angel_utils/conversion.py

```diff
@@ -33,6 +36,18 @@ def time_to_int(time_msg: Time) -> int:
     return (time_msg.sec * SEC_TO_NANO) + time_msg.nanosec
 
 
+def hl2ss_stamp_to_ros_time(timestamp: int) -> Time:
+    """
+    Convert the HL2SS timestamp which is an integer in hundreds of ns to a ROS
+    Time message.
+    """
+    timestamp_sec = timestamp * HUN_NANO_TO_SEC
+    sec = math.floor(timestamp_sec)
+    nanosec = int((timestamp_sec - sec) * SEC_TO_NANO)
```

Comment:
as in `HUN_NANO_TO_SEC` is 1e-7? If I understand correctly, we could simplify some of this math to
```
# Assuming HECTO_NS is defined as a global to int(1e7)
sec = timestamp // HECTO_NS
nanosec = timestamp % HECTO_NS
```